types/src/v2/params.rs

```diff
@@ -89,40 +89,39 @@ impl Serialize for TwoPointZero {
 /// The data containing the params is a `Cow<&str>` and can either be a borrowed `&str` of JSON from an incoming
 /// [`super::request::JsonRpcRequest`] (which in turn borrows it from the input buffer that is shared between requests);
 /// or, it can be an owned `String`.
+///
+/// Regards empty `JSON array` and `JSON object` as no parameters provided.
 #[derive(Clone, Debug)]
-pub struct RpcParams<'a>(Option<Cow<'a, str>>);
+pub struct RpcParams<'a>(Cow<'a, str>);
 
 impl<'a> RpcParams<'a> {
 	/// Create params
 	pub fn new(raw: Option<&'a str>) -> Self {
-		Self(raw.map(Into::into))
+		Self(match raw.unwrap_or("").trim() {
```

Comment:
Moved this check to the `constructor` instead of doing it in every method, also we used `unwrap_or("null")` in `fn parse` but that is not needed anymore.